06.wc/ruby_wc_command.rb

```diff
@@ -0,0 +1,68 @@
+#! /usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  option = ARGV.getopts('l')
+  path_list = ARGV
+  file_info_list = (FileTest.exist?(path_list[0].to_s) ? file_info_list_by_path(path_list) : file_info_list_by_stdin)
+  file_info_list.each { |params| print_format(params, option['l']) }
+  print_total(file_info_list, option['l']) if file_info_list.size > 1
+end
+
+def file_info_list_by_path(path_list)
+  path_list.map do |path|
+    path_text = IO.read(path)
+    create_file_info_list(path_text, path)
+  end
+end
+
+def file_info_list_by_stdin
+  $stdin.tty? ? input_text : pipe_text
+end
+
+def pipe_text
+  pipe_text = $stdin.read
+  [create_file_info_list(pipe_text, nil)]
+end
+
+def input_text
+  stdin_text = []
+  while (line = $stdin.gets)
+    stdin_text << line
+  end
+  [create_file_info_list(stdin_text.join, nil)]
+end
+
+def create_file_info_list(text, path)
+  {
+    line: text.split("\n").size,
+    word: text.split(/\s+/).size,
+    bytesize: text.bytesize,
+    filename: (path || '')
+  }
+end
+
+def print_format(info, option)
+  printf('%<line>8s', info)
+  printf(['%<word>8s', '%<bytesize>8s'].join, info) if option != true
```

Comment:
```suggestion
  printf('%<word>8s%<bytesize>8s', info) if option != true
```

joinを無くした方がシンプルかも？

`if option != true` は冗長なので、`unless option`と書く方が主流です。

https://qiita.com/jnchito/items/dedb3b889ab226933ccf#-true%E3%82%84-false-nil%E3%82%92%E6%98%8E%E7%A4%BA%E7%9A%84%E3%81%AB%E6%9B%B8%E3%81%8B%E3%81%AA%E3%81%84

ただ、`take_a_walk unless raining`みたいに何の条件分岐か想像が付くような名前を付けたいですね。

Reply:
> 仮引数と実引数とで離れた命名というのはありなんでしょうか？

これはありだと思います。
https://bootcamp.fjord.jp/pages/383 の「メソッドの引数も考え方は同じ」に近い話が書いてあります。

> いただいたヒントを参考にoptionをprint_line_onlyとしてみました

`print_line_only`という名前もありといえばありですが、動詞＋名詞なのでぱっと見、メソッドに勘違いされてしまう恐れもあります。

参考
https://qiita.com/jnchito/items/459d58ba652bf4763820#%E5%87%A6%E7%90%86%E3%82%92%E5%AE%9F%E8%A1%8C%E3%81%99%E3%82%8B%E3%83%A1%E3%82%BD%E3%83%83%E3%83%89%E3%81%AF-%E5%8B%95%E8%A9%9E%E3%81%AE%E3%81%BF-or-%E5%8B%95%E8%A9%9E--%E5%90%8D%E8%A9%9E%E3%81%AB%E3%81%99%E3%82%8B

ここは`line_only`だけでもいいかもしれません。